mqtt/mqtt-broker/src/broker.rs

```diff
@@ -109,17 +109,16 @@ where
             .into_iter()
             .flat_map(HashMap::values)
         {
-            if let Ok(client_info) = session.client_info() {
-                let sub_topic_filter = sub.filter().to_string();
-                let operation = Operation::new_subscribe(proto::SubscribeTo {
-                    topic_filter: sub_topic_filter.clone(),
-                    qos: *sub.max_qos(),
-                });
-                activities.push((
-                    client_id.clone(),
-                    Activity::new(client_id.clone(), client_info.clone(), operation),
-                ));
-            }
+            let client_info = session.client_info();
+            let sub_topic_filter = sub.filter().to_string();
+            let operation = Operation::new_subscribe(proto::SubscribeTo {
```

Comment:
this is a bit weird: to create `proto::SubscribeTo` only to later destroy it and create internal representation :(

Reply:
When do we destroy it later?

Reply:
when we convert from proto::SubscribeTo to auth::Subscribe